resreporter/mem_linux.go

```diff
@@ -0,0 +1,70 @@
+package resreporter
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+)
+
+const meminfo = "/proc/meminfo"
+
+var Memory = memory{}
+
+type memory struct{}
+
+func (memory) Used() (used uint64, total uint64) {
+	fd, err := os.Open(meminfo)
+	if err != nil {
+		//XXX Should this use metafora's logger somehow?
+		log.Printf("Error reading free memory via "+meminfo+": %v", err)
```

Comment:
There's two reasons I didn't do that:

1. This should "Never Happen" ... well there's 3 circumstances I can think of: (1) containers/chroots without /proc access, (2) exotic linux's that do strange things, and (3) running out of FDs.
2. What would the code handling these errors look like? It seems like the only sane thing it could do is either disable the balancer or shutdown... we'd have to make that decision for users at a higher level. I don't think a balancer can even cause a shutdown right now.

So I don't know... it mainly doesn't return an error because I couldn't think of anything useful to do with it, so why clutter the API?

I guess returning the error would at least take care of the awkward logging situation...

I'm going to mull it over for a bit, let me know if you have an opinion either way.